libs/settings/api/src/lib/settings.resolver.ts

```diff
@@ -0,0 +1,48 @@
+import {Args, Mutation, Query, Resolver} from '@nestjs/graphql'
+import {CanGetSettings, CanUpdateSettings, Permissions} from '@wepublish/permissions/api'
+import {Setting, UpdateSettingInput, SettingFilter} from './settings.model'
+import {SettingsService} from './settings.service'
+
+@Resolver()
+export class SettingsResolver {
+  constructor(private settingsService: SettingsService) {}
+
+  /*
+  Queries
+ */
+  @Query(returns => [Setting], {
+    name: 'settingsList',
+    description: `
+      Returns all settings.
+    `
+  })
+  @Permissions(CanGetSettings)
+  settingsList(@Args('filter', {nullable: true}) filter?: SettingFilter) {
+    return this.settingsService.settingsList(filter)
+  }
+
+  @Query(returns => Setting, {
+    name: 'setting',
+    description: `
+      Returns a single setting by id.
+    `
+  })
+  @Permissions(CanGetSettings)
+  setting(@Args('id') id: string) {
+    return this.settingsService.setting(id)
+  }
+
+  /*
+  Mutations
+ */
+  @Mutation(returns => [Setting], {
+    name: 'updateSettings',
```

Comment:
Do we want to only support updating 1 setting at the time and let the frontend iterate over it? Makes the API cleaner IMO.

The frontend can batch the requests in GraphQL.

Reply:
that's how it was done in the past, I simply moved it to nest

Reply:
ok @Itrulia I changed the resolver to support 1 setting at the time and I call it in batch on the frontend as suggested. This way I could use the @ArgsType() and it's cleaner